card/README_Developers.rdoc

```diff
@@ -0,0 +1,100 @@
+
+MoFoS stands for "Model-Formats-Sets": One Model, viewable in many Formats, divisible into Sets.
+
+"Everything is a Card" is an idea familiar to any Wagneer. As you may have guessed Cards are Wagn's "one Model" in MoFoS terms.
+
+You may have also noticed that you can add .html (the default), .text, .json, etc to any card's web path to receive the card's contents in a different format. This is the Format in MoFos.
+
+The great power of MoFoS, though, is in the Sets. Thanks to Sets, code rules (and data rules) can be as narrow or specific as desired.  These nonorthogonal groupings open up lots of exciting possibilities, and they're a big part of what makes Wagn so much fun.
+
+
+===Is Wagn Ruby on Rails?
+
+It's likely that many folks reading this are Ruby-on-Rails developers. They'll find lots of familiar patterns in Wagn, and they may have noticed that wagn depends upon rails gems.  We should be clear: <b>Wagn is a Rails descendant and owes a ton to the Rails community, but Wagn is not Rails</b>.
+
+The main difference is that while Rails' core pattern is creating new things (especially models, views, and controllers), Wagn's core pattern is subdividing exising things into new Sets.
+
+Wagn mod developers do create lots of views (though they're quite different from Rails views), but we never create new models or controllers.  In the Wagn gem there is one controller (CardController), one main model ({Card}), and several other models to optimize the Card model and track its history ({Card::Reference}, {Card::Act}, {Card::Action}, and {Card::Change}).
+
+Wagn's equivalent of creating a new model is configuring a new Set of cards. Its equivalent of creating new actions on controllers is creating new events that are conditionally triggeres for a Set of cards.
+
+Rails development experience is definitely a plus in getting to know Wagn, but Wagn offers a new architecture, and lot of new possibilities.
+
+
+==Mods
+
+A Wagn Mod is a discrete piece of Wagn functionality. If you want to customize a Wagn deck in a way that can't be done on the site itself, odds are you want a mod.
+
+The most convenient way to add a mod is in the `mod` directory of your deck, eg:
+
+  /mydeck/mod/mymod
+
+A mod can have several different directories: <i>chunk, format, layout, set, set_pattern, spec, file, and lib</i>. Of these, the set directory is by far the most important.
+
+===Creating a mod with set modules
+
+Set modules are defined in a mod's set directory, eg:
+
+  /mydeck/mod/mymod/set/
+
+For example, if your deck has Company cards, you can extend the behavior of those cards by adding a set module like so:
+
+  /mydeck/mod/mymod/set/type/company.rb
+
+Note that "company" here does not refer to its "name", but rather its "codename" (which an administrator might add to the Company card via the RESTful web API with a url like
+
+  /update/Company?card[codename]=company
+
```

Comment:
We do; they should be in here.  They're probably mature enough for mod developers.  Just haven't gotten to it yet.